src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkWildcardNotDefinedInspection.kt

```diff
@@ -0,0 +1,34 @@
+package com.jetbrains.snakecharm.inspections
+
+import com.intellij.codeInspection.LocalInspectionToolSession
+import com.intellij.codeInspection.ProblemsHolder
+import com.jetbrains.python.psi.PyReferenceExpression
+import com.jetbrains.snakecharm.SnakemakeBundle
+import com.jetbrains.snakecharm.stringLanguage.lang.psi.elementTypes.SmkSLReferenceExpressionImpl
+
+class SmkWildcardNotDefinedInspection : SnakemakeInspection() {
+    override fun buildVisitor(
+            holder: ProblemsHolder,
+            isOnTheFly: Boolean,
+            session: LocalInspectionToolSession
+    ) =
+            object : SmkWildcardsInspectionVisitor<SmkSLReferenceExpressionImpl>(holder, session) {
+        override fun visitPyReferenceExpression(expr: PyReferenceExpression) {
+            if (expr !is SmkSLReferenceExpressionImpl || !expr.isWildcard()) {
```

Comment:
Also here we are using `visitor` design pattern in order not to write
```kotlin
if (element is PyReferenceExpression) {
..
} 
if (element is SmkSLReferenceExpression) {
...
}
if (element is Rule) {
..
}
```
You need here own `visitSmkSLReferenceExpression()` method and it's support in `SmkSLReferenceExpressionImpl.accept()`